File: common/src/main/java/org/tdf/common/trie/Dump.java
package org.tdf.common.trie;

import lombok.AccessLevel;
import lombok.Getter;
import org.tdf.common.util.HexBytes;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

@Getter(AccessLevel.PACKAGE)
class Dump implements ScannerAction {
    private Map<HexBytes, HexBytes> pairs = new HashMap<>();

    @Override
    public Boolean apply(TrieKey path, Node node) {
        if (node.getHash() != null) {
            pairs.put(
                HexBytes.fromBytes(node.getHash()),
                HexBytes.fromBytes(Objects.requireNonNull(node.rlp.getEncoded()))
            );
        }
        return true;
    }
}

File: common/src/main/java/org/tdf/common/trie/Dump.kt
package org.tdf.common.trie

import org.tdf.common.util.HexBytes
import org.tdf.common.util.hex

internal class Dump : ScannerAction {
    internal val pairs: MutableMap<HexBytes, HexBytes> = HashMap()

    override fun scan(path: TrieKey, node: Node): Boolean {
        if (node.hash != null) {
            pairs[node.hash.hex()] = node.rlp.hex()
        }
        return true
    }
}